src/sparseml/recipe_template/utils.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional
+
+from sparseml.pytorch.sparsification import (
+    ACDCPruningModifier,
+    GMPruningModifier,
+    MagnitudePruningModifier,
+    Modifier,
+    QuantizationModifier,
+)
+from sparseml.sparsification import ModifierYAMLBuilder
+
+
+__all__ = [
+    "ModifierBuildInfo",
+]
+
+
+class ModifierBuildInfo:
+    """
+    A class with state and helper methods for building a recipe from
+    modifier(s)
+
+    :param modifier: class of the Modifier in question
```

Comment:
Would be great to have some documentation (pseudocode in this docstring, alternatively unit test that would have a side-effect of being a form of documentation) of how the potential use of this class in code would be.